### PR TITLE
[Clusterloader2] Implement a clusterloader2 tester

### PR DESCRIPTION
**Pull request by amwat**
Create an initial implementation of clusterloader2 tester.
Currently requires having https://github.com/kubernetes/perf-tests checked out.

/cc @BenTheElder

### CODE REVIEW

File: kubetest2-kind/deployer/deployer.go [hunk 1]
```diff
@@ -63,6 +63,17 @@ type deployer struct {
 	verbosity      int    // --verbosity for kind
 }
 
+func (d *deployer) Kubeconfig() (string, error) {
```
Comment (BenTheElder) [inline kubetest2-kind/deployer/deployer.go:66]: this isn't quite correct vs kind / client-go FYI ... https://github.com/kubernetes-sigs/kind/tree/master/pkg/cluster/internal/kubeconfig/internal/kubeconfig

File: pkg/testers/clusterloader2/cl2.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+	"sigs.k8s.io/kubetest2/pkg/exec"
+	suite "sigs.k8s.io/kubetest2/pkg/testers/clusterloader2/suite"
+)
+
+type Tester struct {
+	Suites        string `desc:"Comma separated list of standard scale testing suites e.g. load, density"`
+	TestOverrides string `desc:"Comma separated list of paths to the config override files. The latter overrides take precedence over changes in former files."`
+	TestConfigs   string `desc:"Comma separated list of paths to test config files."`
+	Provider      string `desc:"The type of cluster provider used (e.g gke, gce, skeleton)"`
+	KubeConfig    string `desc:"Path to kubeconfig"`
```
Comment (BenTheElder) [inline pkg/testers/clusterloader2/cl2.go:22]: shouldn't we inject the kubeconfig env to all testers?

Comment (BenTheElder) [reply pkg/testers/clusterloader2/cl2.go:22]: (this also allows us to do multi-file for multi-cluster)

File: kubetest2-kind/deployer/deployer.go [hunk 3]
```diff
@@ -90,8 +101,8 @@ func bindFlags(d *deployer) *pflag.FlagSet {
 	return flags
 }
 
-// assert that deployer implements types.Deployer
-var _ types.Deployer = &deployer{}
+// assert that deployer implements types.DeployerWithKubeconfig
```
Comment (BenTheElder) [inline kubetest2-kind/deployer/deployer.go:104]: Is there any deployer that doesn't have kubeconfig?

File: pkg/testers/clusterloader2/cl2.go (hunk 2 above)
Comment (amwat) [reply pkg/testers/clusterloader2/cl2.go:22]: we already do, this is an override. And yeah clusterloader2 doesn't support multi kubeconfig yet :/

File: kubetest2-kind/deployer/deployer.go (hunk 3 above)
Comment (amwat) [reply kubetest2-kind/deployer/deployer.go:104]: Don't think so

File: kubetest2-kind/deployer/deployer.go (hunk 1 above)
Comment (amwat) [reply kubetest2-kind/deployer/deployer.go:66]: are you referring to https://github.com/kubernetes-sigs/kind/blob/master/pkg/cluster/internal/kubeconfig/internal/kubeconfig/paths.go#L30-L43 ?

I think we are only missing the KUBECONFIG env handling from there. will add as a follow up. This should solve the basic case when nothing is set.

File: kubetest2-kind/deployer/deployer.go (hunk 3 above)
Comment (BenTheElder) [reply kubetest2-kind/deployer/deployer.go:104]: I think we should make this a standard part of the API for deployer, a requirement for kubetest2, but that can be a follow-up.

File: kubetest2-kind/deployer/deployer.go (hunk 1 above)
Comment (BenTheElder) [reply kubetest2-kind/deployer/deployer.go:66]: ~HOME is also more complex on windows, off the top of my head.

I think this deployer should probably have it's own stable computed paths and use those consistently (like $HOME/.kube/kubetest2-kind-$cluster)

File: pkg/testers/clusterloader2/cl2.go (hunk 2 above)
Comment (BenTheElder) [reply pkg/testers/clusterloader2/cl2.go:22]: the usage text should probably clarify that this is an override?

**Comment by BenTheElder** [comment]
/lgtm
/approve
/hold
some nits

File: kubetest2-kind/deployer/deployer.go (hunk 3 above)
Comment (amwat) [reply kubetest2-kind/deployer/deployer.go:104]: ack

File: pkg/testers/clusterloader2/cl2.go (hunk 2 above)
Comment (amwat) [reply pkg/testers/clusterloader2/cl2.go]: done

File: kubetest2-kind/deployer/deployer.go (hunk 1 above)
Comment (amwat) [reply kubetest2-kind/deployer/deployer.go:66]: yeah, I'm planning on sending kind deployer specific changes separately soon and the top-level artifacts organization changes, will fix this more robustly there.

Current change is only to make quick testing of the clusterloader2 tester locally with the kind deployer easy.

**Comment by amwat** [comment]
/hold cancel 
most of the actual kind deployer specific changes will be done in a follow up.
This PR will now enable basic clusterloader2 testing.

**Pull request merged**